cmake/modules/RootBuildOptions.cmake

```diff
@@ -363,14 +363,14 @@ endif()
 #---webgui by default always build together with root7-----------------------------------------
 set(webgui_defvalue ${root7_defvalue})
 
-#---Define at moment the options with the selected default values------------------------------
-ROOT_APPLY_OPTIONS()
-
 #---Enable asserts for Debug builds and for the dev mode---------------------------------------
 if(CMAKE_BUILD_TYPE STREQUAL "Debug" OR dev)
```

Comment:
Do you also need quotes here for non-build-type generators?
if("CMAKE_BUILD_TYPE" STREQUAL "Debug" OR dev)
`

Reply:
Also note that this will fail if somebody configures with
`-DCMAKE_BUILD_TYPE=DEBUG`

Reply:
This can probably fail with multi-configuration generators, as `CMAKE_BUILD_TYPE` is not defined in that case. However, to quote you'd have to use `"${CMAKE_BUILD_TYPE}"`, otherwise you would just get the string `"CMAKE_BUILD_TYPE"` compared and that would always be false. With the variable, I think it probably works because this is a native CMake variable, but it's worth checking anyway. If configuring with `-DCMAKE_BUILD_TYPE=DEBUG` fails as you mentioned, then I at least have a workaround for Gentoo, but in any case, the `if` condition should then be changed so that it uses `MATCHES` and uses tolower/toupper before comparing.

Reply:
If you can take care of that `if` condition, I think we can merge this then. What do you think?